src/cdk/app.ts

```diff
@@ -14,43 +13,140 @@ export class LambdaService extends cdk.Stack {
+
+        const vpcId = new cdk.CfnParameter(this, 'VpcId', {
+            type: 'AWS::EC2::VPC::Id',
+            description: 'VPC in which instances will run',
+        });
+
+        const publicSubnets = new cdk.CfnParameter(this, 'Subnets', {
+            type: 'List<AWS::EC2::Subnet::Id>',
+            description: 'Subnets where instances will run',
+        });
+
+        const availabilityZones = new cdk.CfnParameter(this, 'AZs', {
+            type: 'List<AWS::EC2::AvailabilityZone::Name>',
+            description: 'List of AZs',
+        });
 
-        const handler = new lambda.Function(this, 'frontend-contributions-service', {
-            runtime: lambda.Runtime.NODEJS_12_X,
-            code: lambda.Code.fromBucket(
-                s3.Bucket.fromBucketName(this, 'lambda-code-bucket', bucket),
-                key,
-            ),
-            handler: 'server.handler',
-            functionName,
+        const ami = new cdk.CfnParameter(this, 'AMI', {
+            type: 'AWS::EC2::Image::Id',
+            description: 'AMI ID to be provded by RiffRaff',
         });
 
-        const restApiName = `contributions-service-${stage.value}`;
+        const vpc = ec2.Vpc.fromVpcAttributes(this, 'vpc', {
+            vpcId: vpcId.valueAsString,
+            availabilityZones: availabilityZones.valueAsList,
+            publicSubnetIds: publicSubnets.valueAsList,
+        });
+
+        const role = new iam.Role(this, 'role', {
+            assumedBy: new iam.ServicePrincipal('ec2.amazonaws.com'),
+            inlinePolicies: {
+                instancePolicy: new iam.PolicyDocument({
+                    statements: [
+                        new iam.PolicyStatement({
+                            effect: iam.Effect.ALLOW,
+                            resources: ['arn:aws:s3:::aws-frontend-contributions-service/*'],
+                            actions: ['s3:GetObject', 's3:HeadObject', 's3:List*'],
+                        }),
+                        new iam.PolicyStatement({
+                            effect: iam.Effect.ALLOW,
+                            resources: ['*'],
+                            actions: ['ssmmessages:*', 'ssm:*', 'ec2messages:*', 'logs:*'],
+                        }),
+                    ],
+                }),
+            },
+        });
+
+        const userData = ec2.UserData.forLinux();
+        const App = app.value.toString();
+        const Stack = stack.value.toString();
+        const Stage = stage.value.toString();
+
+        userData.addCommands(
+            `groupadd frontend`,
+            `useradd -r -m -s /usr/bin/nologin -g frontend ${App}`,
+
+            `export App=${App}`,
+            `export Stack=${Stack}`,
+            `export Stage=${Stage}`,
+            `export NODE_ENV=production`,
+            `export BASE_URL=\`aws ssm get-parameter --name "/contributions-service/${Stage.toLowerCase()}/base_url" --region eu-west-1 | jq ".Parameter.Value"\``,
```

Comment:
This env var is required and we attempt to ensure it's been set in the healthcheck to ensure the deploy fails if it isn't set. If anything goes wrong with the `aws` call or the `jq` call here they print to stderr, which means the value of the env var gets set to empty string, so I think that's what we want. Would be good to get some feedback on this if anyone has thoughts.